featurehub_config.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from client_context import ClientContext, ClientEvalFeatureContext, ServerEvalFeatureContext
+from edge_service import EdgeService
+from featurehub_repository import FeatureHubRepository
+
+
+class FeatureHubConfig:
+    # readyness: Readyness;
+
+    _api_keys = []
+    _api_key: str
+    _edge_url: str
+    _client_eval: bool
+
+    def __init__(self, edge_url, api_key):
+        self._repository = FeatureHubRepository
+        self._edge_url = edge_url
+        self._api_key = api_key
+
+        if not edge_url or not api_key:
+            raise TypeError('apiKey and edge_url must not be null')
+
+        self._api_keys = [api_key]
+
+        self._client_eval = '*' in self._api_key
+
+        if not self._edge_url[-1] == '/':
+            self._edge_url += '/'  # think this will fail as strings are immutable
+
+        self._url = self._edge_url + 'features/' + self._api_key
+
+    def api_key(self, api_key: str) -> FeatureHubConfig:
+        self._api_keys.append(api_key)
+        return self
+
+    def client_evaluated(self) -> bool:
+        return self._client_eval  # is this correct?
+
+    def get_api_keys(self) -> list[str]:
+        api_keys = self._api_keys
+        return api_keys
```

Comment:
if you are trying to ensure you return a copy, are you sure this makes a copy? in java for example you need to go api_keys.addAll(self._api_keys)

Reply:
it does that in api_key function. this one just returns _api_keys isn't it?